```
src/headerssync.cpp:316:26: error: passing result of std::move() as a const reference argument; no move will actually happen [performance-move-const-arg,-warnings-as-errors]
    return CBlockLocator{std::move(locator)};
                         ^~~~~~~~~~       ~
```

--- a/src/chain.cpp
+++ b/src/chain.cpp
@@ -49,7 +49,7 @@ std::vector<uint256> LocatorEntries(const CBlockIndex* index)
 
 CBlockLocator GetLocator(const CBlockIndex* index)
 {
-    return CBlockLocator{std::move(LocatorEntries(index))};
+    return CBlockLocator{LocatorEntries(index)};
 }
 
 CBlockLocator CChain::GetLocator() const

--- a/src/primitives/block.h
+++ b/src/primitives/block.h
@@ -123,7 +123,7 @@ struct CBlockLocator
 
     CBlockLocator() {}
 
-    explicit CBlockLocator(const std::vector<uint256>& vHaveIn) : vHave(vHaveIn) {}
+    explicit CBlockLocator(std::vector<uint256>&& have) : vHave(std::move(have)) {}
 
     SERIALIZE_METHODS(CBlockLocator, obj)
     {

